samcli/local/docker/lambda_build_container.py

```diff
@@ -81,14 +83,14 @@ def __init__(  # pylint: disable=too-many-locals
         )
 
         if image is None:
-            image = LambdaBuildContainer._get_image(runtime, architecture)
+            image = LambdaBuildContainer._get_image(runtime, architecture, build_method=build_method)
         entry = LambdaBuildContainer._get_entrypoint(request_json)
         cmd = []
 
         additional_volumes = {
             # Manifest is mounted separately in order to support the case where manifest
             # is outside of source directory
-            manifest_dir: {"bind": container_dirs["manifest_dir"], "mode": "ro"}
+            manifest_dir: {"bind": container_dirs["manifest_dir"], "mode": "rw"}
```

Comment:
I don't know how `additional_volumes` are used. Did your tests actually fail without this change too?